cmd/load-runner/main.go

```diff
@@ -0,0 +1,171 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"os"
+	"os/exec"
+	"os/signal"
+	"regexp"
+	"time"
+)
+
+type config struct {
+	duration       string
+	parsedDuration time.Duration
+	namespace      string
+	loadName       string
+	chartFilesPath string
+}
+
+const (
+	defaultReleaseNamePrefix = "rudder-load"
+	defaultChartFilesPath    = "./artifacts/helm"
+)
+
+func main() {
+	cfg := parseFlags()
+	if err := validateInputs(cfg); err != nil {
+		fmt.Fprintf(os.Stderr, "Validation error: %v\n", err)
+		os.Exit(1)
+	}
+
+	if err := run(cfg); err != nil {
+		fmt.Fprintf(os.Stderr, "Error: %v\n", err)
+		os.Exit(1)
+	}
+}
+
+func parseFlags() *config {
+	var cfg config
+	flag.StringVar(&cfg.duration, "d", "", "Duration to run (e.g., 1h, 30m, 5s)")
+	flag.StringVar(&cfg.namespace, "n", "", "Kubernetes namespace")
+	flag.StringVar(&cfg.loadName, "l", "", "Load scenario name")
+	flag.StringVar(&cfg.chartFilesPath, "f", "", "Path to the chart files (e.g., artifacts/helm)")
+
+	flag.Usage = func() {
+		fmt.Fprintf(os.Stderr, "\nUsage: %s [options]\n\n", os.Args[0])
+		fmt.Fprintf(os.Stderr, "Options:\n")
+		flag.PrintDefaults()
+		fmt.Fprintf(os.Stderr, "\nExamples:\n")
+		fmt.Fprintf(os.Stderr, "  %s -d 1m -n test -l test-staging    # Run test-staging load for 1 minute on test namespace\n", os.Args[0])
+	}
+
+	flag.Parse()
+
+	if cfg.duration == "" || cfg.namespace == "" || cfg.loadName == "" {
+		if cfg.duration == "" {
+			fmt.Fprintf(os.Stderr, "Error: duration is required\n")
+		}
+		if cfg.namespace == "" {
+			fmt.Fprintf(os.Stderr, "Error: namespace is required\n")
+		}
+		if cfg.loadName == "" {
+			fmt.Fprintf(os.Stderr, "Error: load name is required\n")
+		}
+
+		flag.Usage()
+		os.Exit(1)
```

Comment:
Can we please define one pattern and stick with it? I think it makes sense to have an exit in `main` so I'd try to avoid spreading them. Perhaps this function could simply return an error. Also it can be invoked by `run` after all so you exit only once in `main`.

Reply:
done